### PR TITLE
API-compatible interface to a new combined suburbs/all_suburbs/result JSON files

### PR DESCRIPTION
**Pull request by lyricnz**
No improvements, but switch to the combined file, and generate "old format" results for API compatibility.   Should probably stop auto-processing, then update this (and remove old files), before merging.

### CODE REVIEW

**Comment by lyricnz** [comment]
Combined results/suburb file looks like 

```json
{
    "ACT": [
        {
            "name": "Acton",
            "state": "ACT",
            "processed_date": "2023-07-07T03:54:25.154530",
            "announced": false,
            "announced_date": null
        },
        {
            "name": "Ainslie",
            "state": "ACT",
            "processed_date": "2023-06-05T19:37:43.447130",
            "announced": true,
            "announced_date": "November 2024"
        },
        {
            "name": "Amaroo",
            "state": "ACT",
            "processed_date": "2023-07-07T04:05:37.940829",
            "announced": false,
            "announced_date": null
        },
```

**Comment by lyricnz** [comment]
Most of the diff is moving the json utility functions. I'll PR that separately to reduce noise.

**Comment by lyricnz** [comment]
Suggest merge https://github.com/LukePrior/nbn-upgrade-map/pull/171 first

**Comment by lyricnz** [comment]
I think this is ready (needs pause, rerun/update the adhoc script here, delete old files, then merge).  

Future steps:
- more accurate timestamps in results
- some things other people have suggested

**Comment by LukePrior** [comment]
Alright cool,

> Combined results/suburb file looks like

```json
"ACT": [
  {
    "name": "Acton",
    "state": "ACT",
    "processed_date": "2023-07-07T03:54:25.154530",
    "announced": false,
    "announced_date": null
  }
]
```

Is it worth including the state for each suburb if they are already grouped by state?

I'll hopefully update the website tomorrow to use this file and it can be merged in.

**Comment by lyricnz** [comment]
> Is it worth including the state for each suburb if they are already grouped by state?

We can remove it from the file (and repopulate it during load if you like).  I only kept it because the existing results.json had it.

```python
class Suburb:
    name: str
    state: str  # redundant but useful
    processed_date: datetime = None
    announced: bool = False  # should be redundant vs announced_date, but isn't
    announced_date: str = None  # TODO: datetime?
```

> I'll hopefully update the website tomorrow to use this file and it can be merged in.

Ah I didn't spot that the website used this file.  That actually uses suburb.state at the moment. Suggest add your changes to this PR (so we can test etc)

**Comment by lyricnz** [comment]
Here's my attempt to update the index.html (using the PR json file - will need to be updated for merge)

https://github.com/LukePrior/nbn-upgrade-map/pull/168/commits/5d1c8d3e82667bc918b0be60ba5135621cb3da7d

Seems to work.  I made a couple of local variables to simplify the <option> construction

**Comment by lyricnz** [comment]
JSON file now looks like
```json
{
 "ACT": [
  {
   "name": "Acton",
   "processed_date": "2023-07-07T03:54:25.154530",
   "announced": false,
   "announced_date": null
  },
  {
   "name": "Ainslie",
   "processed_date": "2023-06-05T19:37:43.447130",
   "announced": true,
   "announced_date": "November 2024"
  },
  {
   "name": "Amaroo",
   "processed_date": "2023-07-07T04:05:37.940829",
   "announced": false,
   "announced_date": null
  },
...
```

File: code/data.py [hunk 1]
```diff
@@ -49,13 +49,23 @@ class Address:
...
+
+    @property
+    def internal(self):
+        return self.name.upper().replace(" ", "-")
+
+    @property
+    def file(self):
+        return self.name.lower().replace(" ", "-")
+
+    def __eq__(self, other):
```
Comment (LukePrior) [inline code/data.py:64]: Does this need to be updated now that states been removed?

Comment (lyricnz) [reply code/data.py:64]: You're right. Removed.

**Comment by lyricnz** [comment]
When you think the PR is ready:
- remove old json files
- remove old_*() API methods
- remove tests that use the old methods
- update the URL in index.html to point at the real/master URL
- merge, deploy website.

**Comment by lyricnz** [comment]
Future work:
- use Suburb dataclass not the mangled dict in code; including type hints etc
- including more accurate .date field

The reason I split this into two parts is that this first one maintains API compatibility to the rest of the codebase.

**Comment by LukePrior** [comment]
Looks good, Actions has been paused so ready to rerun.

**Comment by lyricnz** [comment]
Pushed those changes

**Comment by LukePrior** [comment]
Perfect!

**Comment by LukePrior** [comment]
Actions seems to keep on redoing Canberra Airport

**Pull request merged**